Raven.Voron/Voron/Platform/Win32/Win32MemoryMapPager.cs

```diff
@@ -363,15 +361,36 @@ public override void MaybePrefetchMemory(List<Page> sortedPages)
 
             fixed (Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY* entries = list.ToArray())
             {
-                Win32MemoryMapNativeMethods.PrefetchVirtualMemory(Win32NativeMethods.GetCurrentProcess(),
+                Win32MemoryMapNativeMethods.PrefetchVirtualMemory(_currentProcess,
                     (UIntPtr)list.Count,
                     entries, 0);
             }
         }
 
+        public override void MaybePrefetchMemory(List<long> pagesToPrefetch)
+        {
+            if (CanPrefetch == false)
+                return; // not supported
+
+            if (pagesToPrefetch.Count == 0)
+                return;
+
+            var entries = new Win32MemoryMapNativeMethods.WIN32_MEMORY_RANGE_ENTRY[pagesToPrefetch.Count];
+            for (int i = 0; i < entries.Length; i++)
+            {
+                entries[i].NumberOfBytes = (IntPtr)(4 * PageSize);
```

Comment:
PrefetchVirtualMemory is an strange beast. It is a hint to the kernel to prefetch some memory (as he sees fit). Therefore, PrefetchVirtualMemory is already doing the work of figuring out if he has to merge or not the requests from multiple threads to do that... they essentially go into a queue and the kernel will decide how best to accomplish it. Merging in user time is essentially wasting resources as the kernel has to do that anyways.